Composer/packages/lib/code-editor/src/BaseEditor.tsx

```diff
@@ -182,18 +182,16 @@ const BaseEditor: React.FC<BaseEditorProps> = (props) => {
     }
   }, [onChange, editor]);
 
-  const errorMsgFromDiagnostics = useMemo(() => {
-    const errors = findErrors(diagnostics);
-    return errors.length ? combineSimpleMessage(errors) : '';
+  const { errorMsgFromDiagnostics, warningMsgFromDiagnostics } = useMemo(() => {
+    const { errors, warnings } = findProblems(diagnostics);
+    return {
+      errorMsgFromDiagnostics: errors.length > 0 ? combineSimpleMessage(errors) : null,
+      warningMsgFromDiagnostics: warnings.length > 0 ? combineSimpleMessage(warnings) : null,
+    };
   }, [diagnostics]);
 
-  const warningMsgFromDiagnostics = useMemo(() => {
-    const warnings = findWarnings(diagnostics);
-    return warnings.length ? combineSimpleMessage(warnings) : '';
-  }, [diagnostics]);
-
-  const hasError = !!errorMessage || !!errorMsgFromDiagnostics;
-  const hasWarning = !!warningMessage || !!warningMsgFromDiagnostics;
+  const hasError = errorMessage ?? errorMsgFromDiagnostics;
```

Comment:
From the variable name it looks like its expecting a boolean right?
shouldn't this be !!(errorMessage || warningMsgFromDiagnostics).

For ex
'abc' ?? undefined = 'abc'
!!('abc' || undefined) = true - We want the value to be represented as boolean.

Am i missing something here?